src/io/msh_vtkreader.cc

```diff
@@ -525,32 +520,25 @@ void VtkMeshReader::create_node_element_caches() {
 		return;
 	}
 
-	ASSERT( !has_compatible_mesh_ ).error();
-
 	has_compatible_mesh_ = true;
 
 	// read Points data section
 	HeaderQuery header_params("Points", 0.0, OutputTime::DiscreteSpace::MESH_DEFINITION);
 	auto point_header = this->find_header(header_params);
-	bulk_elements_id_.resize(point_header.n_entities);
-	for (unsigned int i=0; i<bulk_elements_id_.size(); ++i) bulk_elements_id_[i]=i;
-	this->get_element_data<double>(point_header.n_entities, point_header.n_components, false, 0 );
+	this->get_element_data<double>(point_header, point_header.n_entities, point_header.n_components, false);
 
 	// read offset data section
 	HeaderQuery offsets_params("offsets", 0.0, OutputTime::DiscreteSpace::MESH_DEFINITION);
     auto offset_header = this->find_header(offsets_params);
-    for (unsigned int i=bulk_elements_id_.size(); i<offset_header.n_entities; ++i) bulk_elements_id_.push_back(i);
-    std::vector<unsigned int> &offsets_vec = *( this->get_element_data<unsigned int>(offset_header.n_entities, offset_header.n_components, false, 0 ) );
+    std::vector<unsigned int> &offsets_vec = *( this->get_element_data<unsigned int>(offset_header, offset_header.n_entities, offset_header.n_components, false) );
 
     // read connectivity data section
     HeaderQuery con_params("connectivity", 0.0, OutputTime::DiscreteSpace::MESH_DEFINITION);
-    auto & con_header = this->find_header(con_params);
+    auto con_header = this->find_header(con_params);
     con_header.n_entities = offsets_vec[offsets_vec.size()-1];
-    for (unsigned int i=bulk_elements_id_.size(); i<con_header.n_entities; ++i) bulk_elements_id_.push_back(i);
-    this->get_element_data<unsigned int>(con_header.n_entities, con_header.n_components, false, 0 );
+    this->get_element_data<unsigned int>(con_header, con_header.n_entities, con_header.n_components, false);
 
     has_compatible_mesh_ = false;
-	bulk_elements_id_.clear();
```

Comment:
this part also seems to me like some dead code...
the assert and then setting `has_compatible_mesh_` to true does no make sense.
it seems to me, that `bulk_elements_id_` is not used anywhere for something useful (definitely in this function)..
then it seems that `has_compatible_mesh_` is also useless
and `find_header` and `get_element_data` can be merge together...